shared_model/interfaces/common_objects/impl/ordering.cpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <algorithm>
+#include <cstring>
+
+#include "interfaces/common_objects/ordering.hpp"
+
+using namespace shared_model::interface;
+
+Ordering::Ordering() {
+  reset();
+}
+
+Ordering::Ordering(Ordering&& c) {
+  copy_(c);
+}
+
+Ordering::Ordering(Ordering const& c) {
+  copy_(c);
+}
+
+void Ordering::copy_(Ordering const& src) {
+  static_assert(1 == std::is_pod<typename std::decay<decltype(ordering_[0])>::type>::value, "OrderingEntry must be POD!");
+
+  std::memcpy(ordering_, src.ordering_, sizeof(ordering_));
+  count_ = src.count_;
+}
+
+void Ordering::insert(Field field, Direction direction, PriorityType priority) {
+  auto& entry = ordering_[(size_t)field];
+  if (priority > entry.priority) {
+    count_ += (entry.priority == 0 ? 1 : 0);
+
+    entry.priority = priority;
+    entry.field = field;
+    entry.direction = direction;
+  }
+}
+
+void Ordering::reset() {
+  static_assert(1 == std::is_pod<typename std::decay<decltype(ordering_[0])>::type>::value, "OrderingEntry must be POD!");
```

Comment:
i'm curious why so?
also, `== 1` seems strange, when the standard says the `value` is `true`/`false`.

Reply:
1 is bad, agree. But static_assert is neede here because memset and memcpy usage. It makes impossible to insert non-POD type by other developer whick can cause a bug.